### DISCUSSION TITLE
What are the best parameters that can be used in lettuce here:

### DISCUSSION THREAD

**Discussion by PedroMPagani** [redis/lettuce#2109] [Q&A] [answered]
I'd like to know if there is currently anything I could change in the lettuce redis client parameters in order to improve the performance:

I'm currently using these methods here:

# This is to initialize the client
```java 
        redisClient = RedisClient.create(RedisURI.builder().withHost(host).withPort(port)
                .withPassword(password).withDatabase(0)
                .withTimeout(Duration.ofSeconds(30000)).build());
        sender = redisClient.connect(new ByteArrayCodec());
```

# This is to publish messages ( I know i could do this with pipelines, just don't know how to 

```java 
  public void publishData(byte[] channel,byte[] message){
        Executor.submit(()-> sender.sync().publish(channel,compress(message)));
    }
```

# Registering my pubsub
```java
    public void registerPubSub(RedisPubSubAdapter<byte[],byte[]> sub, byte[] ... names){
        Thread thread = new Thread(()-> {
            StatefulRedisPubSubConnection<byte[],byte[]> connection = redisClient.connectPubSub(new ByteArrayCodec());
            connection.addListener(sub);
            RedisPubSubAsyncCommands<byte[],byte[]> pubSubCommands = connection.async();
            pubSubCommands.subscribe(names);
        });
        pubsubs.add(thread);
        thread.setDaemon(true);
        thread.start();
    }
```

The pipeline usage would mostly just improve the time response of the thread and not blocking it until the response arrives, correct? (Are there any other improvements else than that?)

**Comment by mp911de** [comment, marked as answer]
Subscribing with Lettuce doesn't require an additional thread as the `subscribe` method doesn't block.

Calling 

```java
RedisPubSubAsyncCommands<byte[],byte[]> pubSubCommands = connection.async();
pubSubCommands.subscribe(names)
```

on the main thread is fully sufficient. Other than that, you can call `sender.async().publish(channel,compress(message))` to not await completion of the `PUBLISH` command (fire&forget).
